aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/UploadOptions.java

```diff
@@ -0,0 +1,124 @@
+package com.amazonaws.mobileconnectors.s3.transferutility;
+
+import com.amazonaws.services.s3.model.CannedAccessControlList;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+
+/**
+ * A class that can hold all of the optional parameters (bucket, metadata, cannedAcl, transferListener).
+ */
+public class UploadOptions {
+    private String bucket;
+    private ObjectMetadata metadata;
+    private CannedAccessControlList cannedAcl;
+    private TransferListener listener;
+
+    /**
+     * Constructor.
+     *
+     * @param bucket        The bucket name
+     * @param metadata      The S3 metadata to associate with this object
+     * @param cannedAcl     The canned ACL to associate with this object
+     * @param listener      The listener to attach to transfer observer
+     */
+    public UploadOptions(String bucket, ObjectMetadata metadata, CannedAccessControlList cannedAcl, TransferListener listener) {
```

Comment:
Typically with the Builder pattern, this constructor should take a `Builder` object, rather than the list of parameters.   By doing that, this constructor will never grow to have a super long list of parameters that's hard to maintain.